lib/omniauth/strategies/openid_connect.rb

```diff
@@ -109,7 +111,8 @@ def callback_phase
 
         raise CallbackError, 'Invalid state parameter' if invalid_state
 
-        return fail!(:missing_code, OmniAuth::OpenIDConnect::MissingCodeError.new(params['error'])) unless params['code']
+        grant_type_errors = validate_grant_type_credentials
+        return fail!(*grant_type_errors) if grant_type_errors.present?
```

Comment:
Sorry, I don't see what's the point to build such complex validation: build an array of errors, then return optional `Array`, splat the array and raise the exceptions. Are you sure this is the easiest way to validate credentials?

I'd prefer to call something like `#validate!` and raise an error inside it, instead of this mess.

Btw, it looks like your validation method supports more than one exception in array. But it will raise just first. Very bad.

Reply:
It doesn't really matter at this moment , but yeah, I think I'll remove this`validate_grant_type_credentials` , remove support for the combined `id_token token` response type and just focus on making it working with one type, `id_token` in addition to the existing mode